vault/resource_secrets_sync_config.go

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/vault/helper/namespace"
+)
+
+const (
+	fieldDisabled      = "disabled"
+	fieldQueueCapacity = "queue_capacity"
+)
+
+var syncConfigFields = []string{
+	fieldDisabled,
+	fieldQueueCapacity,
+}
+
+func secretsSyncConfigResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: provider.MountCreateContextWrapper(secretsSyncConfigWrite, provider.VaultVersion116),
+		UpdateContext: secretsSyncConfigWrite,
+		ReadContext:   secretsSyncConfigRead,
+		DeleteContext: secretsSyncConfigDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			fieldDisabled: {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```

Comment:
The request on the Vault side is implemented as a PATCH. If we remove the default value from the TF schema, omitting a field in the resource definition keep it as is instead of reverting to Vault's default. This may be ok if we document it accordingly but most often I've seen that removing an optional field in a TF resource should update it to some default value.

For example, if the current config is:
```
    "data": {
        "disabled": false,
        "queue_capacity": 1234
    },
```

This TF resource definition resets it to 1,000,000:
```
resource "vault_secrets_sync_config" "cfg" {
  disabled       = false
}
```

If we remove the default value here, the same TF resource definition has no effect instead because Vault is called an empty `queue_capacity` field.

Does that make sense? Are we fine with the behavior of not updating to a default value for a missing field?